app/models/manageiq/providers/hawkular/inventory/parser/middleware_manager.rb

```diff
@@ -133,13 +119,10 @@ def uuid_provider_types
       end
 
       def fetch_server_entities
-        @data[:middleware_deployments] = []
-        @data[:middleware_datasources] = []
-        @data[:middleware_messagings] = []
-        @eaps.map do |eap|
-          @ems.child_resources(eap.path, true).map do |child|
+        persister.middleware_servers.each do |eap|
+          collector.child_resources(eap.ems_ref, true).map do |child|
             next unless child.type_path.end_with?('Deployment', 'Datasource', 'JMS%20Topic', 'JMS%20Queue')
-            server = @data_index.fetch_path(:middleware_servers, :by_ems_ref, eap.path)
+            server = persister.middleware_servers.find(eap.ems_ref)
```

Comment:
could we do only find_or_build, if you want to change the data?

the find and lazy_find are reserved for references only now, since for targeted refresh, these can got to the DB to fetch the reference. And there is currently not a strategy that would allow to take the record from the DB. modify it and save. Maybe we can make one up, if that would be the usecase.

So use only find_or_build, if you want to change and save the data, you can probably add there a TODO that it should only find and not build, in that case we would need a condition if the server is not found.